### ISSUE TITLE
Keyboard navigation: missing or inconsistent focus styles, some elements cannot be focused at all

### ISSUE THREAD

**Issue by angelikatyborska** [elixir-lang/ex_doc#1747]
Hi! First of all, let me start by saying how much I love the that all Elixir libraries have their documentation on one website, and they all have the same great looking UI 🙂 

I was browsing around with a keyboard only and noticed a few small problems around that. Let me list them all, from most important to least important. I would love to help out with all of them it that's ok with you.

I've done all my testing in Firefox on macOS, the results might be slightly different in Chrome or Safari.

I have made screenshots for everything, but I am not including them in my original post because it would be very hard to read. Let me know if any of the scenarios I described is unclear and needs a screenshot.

## Some elements cannot be focused at all

- The "copy" button on code snippets if you're not already hovering with your mouse over it.
- The button that closes and opens the left-side menu (this might be on purpose though, see below).

## Some element can be focused even though they aren't visible

Everything in the left-side menu when it's collapsed. There's a funny side effect when it's closed but you still can focus search, which causes the icon to move back and forth horizontally.

It might be on purpose that the left-side menu is never hidden from keyboard navigation (that would explain why you cannot focus the open/close button), but it would need to be fixed to open itself as soon as something inside gets focus.

## Some elements have no visible focus style

- The library version select.
- The theme select in settings modal. Those two are peculiar because if I force focus-visible state on the selects, they do have a visible outline. But it doesn't show when I actually navigate with the keyboard.
- Collapsed links to pages with sections. Only when collapsed.
- The "Search HexDocs" button in the footer.

## Some elements have a different focus style than most of them

Most of the elements have the default user agent focus style, which in my specific case is a 2px wide pink outline. But some have the outline cut off due to `overflow: hidden`, and some have explicitly set a dotted thin outline that's less visible than my system default.

Elements with different focus styles:
- The "settings" button at the top left of the page (thin dotted outline).
- The "copy" button on code snippets (thin dotted outline).
- Expanded links to pages with sections (only the bottom part of the outline shows).
- Links to page sections (only left and right outlines show).
- Expanded links to sections (e.g. "Functions" in modules) (top part of the outline does not show).
- Tab and close buttons in the settings modal.

**Comment by josevalim** [comment]
Hi @angelikatyborska! We probably have never focused (no pun intended) on the focus style of elements, which is why I am not surprised it is everywhere. We would definitely appreciate help and pull requests for those issues. I assume those can be very important for usability too, so it is very welcome!

**Comment by angelikatyborska** [comment]
Great, I'll work on this on the next weekend
